### PR TITLE
@inline(__always) should not imply @inlinable

### PR DESCRIPTION
**Pull request by slavapestov**
This is a request from the standard library team. They want to force inlining of functions inside the standard library without publishing their body for inlining across module boundaries.

I added `@inlinable` in a few places in the stdlib so that this PR has no functional effect (unless I missed something). An audit should be done to remove `@inlinable` from `@inline(__always)` functions where desired.

Fixes <rdar://problem/44657000>, <https://bugs.swift.org/browse/SR-3624>, <rdar://problem/31865137>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please benchmark

**Review by jrose-apple — APPROVED** [review]
Ooops, I thought it already was this way. Thanks, Slava!

**Comment by slavapestov** [comment]
Oops, missed a few.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please benchmark

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please benchmark

**Comment by slavapestov** [comment]
@swift-ci Please benchmark

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/StringGutsRangeReplaceable.swift [hunk 1]
```diff
@@ -40,7 +40,7 @@ extension _StringGuts {
     }
   }
 
-  @usableFromInline // @testable
+  @inlinable // @testable
```
Comment (milseman) [inline stdlib/public/core/StringGutsRangeReplaceable.swift]: I don't think we want `@inlinable` here.

File: stdlib/public/core/StringCharacterView.swift [hunk 2]
```diff
@@ -50,6 +50,7 @@ extension String: BidirectionalCollection {
   }
 
   /// The number of characters in a string.
+  @inlinable
```
Comment (milseman) [inline stdlib/public/core/StringCharacterView.swift]: I don't think we need this to be inlinable for performance

**Review by milseman — COMMENTED** [review]
Great to see this!

I have two String removals: `String.count` and `_StringGuts.isUniqueNative`.

**Comment by slavapestov** [comment]
@milseman Do you mind making the removals yourself in a follow-up PR? I wanted this to be NFC

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by milseman** [comment]
Removed in https://github.com/apple/swift/pull/20601

**Pull request merged**